src/ImageSharp/MetaData/Profiles/Exif/ExifProfile.cs

```diff
@@ -233,8 +232,10 @@ public void SetValue(ExifTag tag, object value)
         /// <summary>
         /// Converts this instance to a byte array.
         /// </summary>
+        /// <param name="includeExifIdCode">Indicates, if the Exif ID code should be included.
+        /// This Exif ID code should not be included in case of PNG's. Defaults to true.</param>
         /// <returns>The <see cref="T:byte[]"/></returns>
-        public byte[] ToByteArray()
+        public byte[] ToByteArray(bool includeExifIdCode = true)
         {
             if (this.values == null)
```

Comment:
You should also check if `this.data` starts with the exif id and skip the first 6 bytes when includeExifIdCode is true.

Reply:
good point, i missed that one. I have added a unit test for that case.